pyro/distributions/multivariate_normal.py

```diff
@@ -0,0 +1,144 @@
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+import warnings
+import torch
+from torch.autograd import Variable
+
+from pyro.distributions.util import copy_docs_from
+from pyro.distributions.distribution import Distribution
+
+
+def potri_compat(var):
+    return Variable(torch.potri(var.data)) if torch.__version__ < '0.3.0' else torch.potri(var)
+
+
+def potrf_compat(var):
+    return Variable(torch.potrf(var.data)) if torch.__version__ < '0.3.0' else torch.potrf(var)
+
+
+def linear_solve_compat(matrix, matrix_chol, y):
+    """Solves the equation ``torch.mm(matrix, x) = y`` for x."""
+    assert matrix.requires_grad == matrix_chol.requires_grad
+    if matrix.requires_grad or y.requires_grad:
+        # If derivatives are required, use the more expensive gesv.
+        return torch.gesv(y, matrix)[0]
+    else:
+        # Use the cheaper Cholesky solver.
+        return torch.potrs(y, matrix_chol)
+
+
+def matrix_inverse_compat(matrix, matrix_chol):
+    """Computes the inverse of a positive semidefinite square matrix"""
+    assert matrix.requires_grad == matrix_chol.requires_grad
+    if matrix.requires_grad:
+        # If derivatives are required, use the more expensive inverse.
+        return torch.inverse(matrix)
+    else:
+        # Use the cheaper Cholesky based potri.
+        return potri_compat(matrix_chol)
+
+
+@copy_docs_from(Distribution)
+class MultivariateNormal(Distribution):
+    """
+    Multivariate normal (Gaussian) distribution.
+
+    A distribution over vectors in which all the elements have a joint
+    Gaussian density.
+
+    :param torch.autograd.Variable loc: Mean. Must be a vector (Variable containing a 1d Tensor).
```

Comment:
nit: Could you ensure that docstrings are at most 80 characters wide (here and in .sample() below)? This is important for users of Jupyter Notebooks so that `help(MultivariateNormal)` prints correctly.